lib/serializer.ts

```diff
@@ -232,7 +232,7 @@ export class Serializer {
         }
         value = this.dateToUnixTime(value);
       } else if (typeName.match(/^TimeSpan$/ig) !== null) {
-        if (!isDuration(value)) {
+        if (!(isDuration(value) || (value.constructor && value.constructor.name === "Duration" && (value as any).isValid && (value as any).isValid()))) {
```

Comment:
Do you need to check if isValid is a function? I'm sure you're fine not checking, but it isn't a function, will this throw an exception when you try to invoke it?

Reply:
Can go with `typeof (value as any).isValid === 'function'`

Reply:
I wish the moment type definitions simply included this member on Durations, maybe will just augment the type definition here locally

Reply:
Done @daschult

Reply:
Cool. I'm glad that worked out well.